server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -336,10 +336,13 @@ public QueryBuilder postFilter() {
 
     /**
      * From index to start the search from. Defaults to {@code 0}.
+     * A value of -1 is interpreted as an unset value equivalent to 0.
+     * Any other negative value is forbidden.
      */
     public SearchSourceBuilder from(int from) {
-        if (from < 0) {
-            throw new IllegalArgumentException("[from] parameter cannot be negative");
+        // accept -1 as the default unset value without error
```

Comment:
Can we set the default early so that we can refuse any negative values ?

Reply:
I think for `SearchSourceBuilder` I would prefer to keep the notion of an "unset" value and interpret that in any way necessary later. I think this class should represent the user request body, e.g. in `innerToXContent` we only output values that are explicitely set by the user. Maybe @javanna has some preferences there as well?